x-pack/plugins/observability_solution/observability_shared/public/components/feature_feedback_button/feature_feedback_button.tsx

```diff
@@ -29,21 +29,28 @@ const getDeploymentType = (isCloudEnv?: boolean, isServerlessEnv?: boolean): str
 const getMLJobType = (mlJobType: NodeType) =>
   mlJobType === 'pod' ? 'Pod Anomalies' : 'Host Anomalies';
 
-const getSurveyFeedbackURL = ({
+export const getSurveyFeedbackURL = ({
   formUrl,
   formConfig,
   kibanaVersion,
-  deploymentType,
   sanitizedPath,
-  mlJobType,
+  isCloudEnv,
+  isServerlessEnv,
+  nodeType,
 }: {
   formUrl: string;
   formConfig?: FormConfig;
   kibanaVersion?: string;
   deploymentType?: string;
   sanitizedPath?: string;
   mlJobType?: string;
+  isCloudEnv?: boolean;
+  isServerlessEnv?: boolean;
+  nodeType?: NodeType;
 }) => {
+  const deploymentType = getDeploymentType(isCloudEnv, isServerlessEnv);
```

Comment:
@jennypavlova I changed the logic here. 

Before it was:
```
const deploymentType =
    isCloudEnv !== undefined || isServerlessEnv !== undefined
      ? getDeploymentType(isCloudEnv, isServerlessEnv)
      : undefined;
```

Because I noticed, while writing some unit tests, that the `Self-Managed (you manage)` option was never reached.

Reply:
> Because I noticed, while writing some unit tests, that the Self-Managed (you manage) option was never reached.

Interesting, I see this option prefilled when I don't have `xpack.cloud.id` in my `kibana.yml` and both values are `false`: Maybe the config you use is different 🤔 If we don't check those values and they are missing because of the cloud plugin response or they are not passed at all because they are optional we then prefill the self-managed as default (and it could be that it's a serverless/cloud env but those values are missing/not passed at all) I would rather leave it empty if we don't have those params then prefill self-managed. Please let me know if I can help you test that.
Side note: when changing this component please check the ML local copy and keep it up to date https://github.com/elastic/kibana/blob/bb5418408eec560afe6a584a443dde2af4d96dc5/x-pack/plugins/ml/public/application/components/feedback_button/feature_feedback_button.tsx

Reply:
Alright that makes sense, I'm reverting my change and fixing the tests.